src/unix/tcp_server.ml

```diff
@@ -95,19 +97,21 @@ let handle_client_async_ (self : t) client_sock client_addr : unit =
     match Encoding.of_int32_le magic_number with
     | Some e -> e
     | None ->
-      Error.failf ~kind:Errors.protocol "Rpc_conn: invalid magic number %ld"
+      Error.failf ~kind:Errors.protocol
+        "RPC server: invalid magic number %ld (when reading the encoding)"
         magic_number
   in
 
   (* spawn a background thread, using the same [active] so as
      to propagate cancellation to it *)
-  let rpc_conn : Rpc_conn.t =
-    Rpc_conn.create ~active:self.active ~buf_pool:self.buf_pool
-      ~server_state:self.st ~runner:self.runner ~encoding ~timer:self.timer ~ic
-      ~oc ()
+  let rpc_conn : conn =
+    Server.For_client.create ~active:self.active ~buf_pool:self.buf_pool
+      ~state:self.st ~runner:self.runner ~encoding ~timer:self.timer ~ic ~oc ()
   in
+  (* TODO: use a fiber isntead? *)
+  ignore (Thread.create Server.For_client.run rpc_conn : Thread.t);
```

Comment:
Sure we want to ignore thread creation failure here?

Reply:
Thread creation failure would just raise. What we're ignoring is the result thread which runs in the background.